repose-aggregator/components/filters/body-patcher-filter/src/main/scala/org/openrepose/filters/bodypatcher/BodyPatcherFilter.scala

```diff
@@ -120,14 +119,9 @@ class BodyPatcherFilter @Inject()(configurationService: ConfigurationService)
   }
 
   def determineContentType(contentType: String): ContentType = {
-    //magic code from stack overflow, i'm a terrible person, http://stackoverflow.com/questions/4636610/how-to-pattern-match-using-regular-expression-in-scala
-    implicit class Regex(sc: StringContext) {
-      def r = new util.matching.Regex(sc.parts.mkString, sc.parts.tail.map(_ => "x"): _*)
-    }
-
     Option(contentType).getOrElse("").toLowerCase match {
-      case r".*json.*" => Json
-      case r".*xml.*" => Xml
+      case Json.regex() => Json
+      case Xml.regex() => Xml
```

Comment:
Honestly, I don't think this looks any worse than it did before. I also appreciate not having to figure out the magics. I mean, it was obvious what the match was doing before, just not how.